packages/ui-components/src/components/QrCode/QrCodeImage.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { Box } from '@material-ui/core';
+import { drawQrCodeCanvas } from './useQrCodeCanvas';
+
+const StyledCanvas = styled.canvas`
+  outline: 1px solid #dedede;
+`;
+
+interface QrCodeImageProps {
+  qrCodeContents: string;
+  humanReadableId: string;
+  width?: number | undefined;
+  margin?: number | string | undefined;
+}
+
+export const QrCodeImage = ({ qrCodeContents, humanReadableId, width = undefined, margin = 'auto' }: QrCodeImageProps) => {
+  const ref = useRef<HTMLCanvasElement | null>(null)
+  useEffect(() => {
+    drawQrCodeCanvas(ref, humanReadableId, qrCodeContents);
+  }, [ref]);
+
+
+  return (
+    <Box style={{
+      display: 'flex',
```

Comment:
Can we make a styled component for this, instead of inline styles? This way it matches the pattern we have already

Reply:
This one is reliant on a prop from the component, so is there a way to do that?

Reply:
Yep, you would pass the styled component props of `width` and `margin` and then use those in tge styled component, similar to how you use themes in styled components.
E.g. 
``` width: ${(props) => props.width};```